### PR TITLE
CC |  Dragonball: add confidential_vm_type for TDX

### PR DESCRIPTION
**Pull request by studychao**
add confidential_vm_type to let Dragonball create a TDX VM.

Signed-off-by: fengshifang <fengshifang@linux.alibaba.com>
Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

### CODE REVIEW

File: src/dragonball/src/kvm_context.rs [hunk 1]
```diff
@@ -91,6 +92,18 @@ mod x86_64 {
     use std::collections::HashSet;
 
     impl KvmContext {
+        /// Create a virtual machine object witch type.
```
Comment (fidencio) [inline src/dragonball/src/kvm_context.rs:95]: s/witch/which

**Review by fidencio — COMMENTED** [review]
lgtm, apart from a possible typo in one comment.

TDX CIs are failing due to network issues to download the crates needed to build the runtime-rs.  It's important to also mention that while we're building the runtime-rs, we're not yet testing the runtime-rs with TDX, and that should be a follow-up work from your side (that I can guide you through, of course).

**Comment by fidencio** [comment]
```
    Checking dragonball v0.1.0 (/home/runner/work/kata-containers/kata-containers/src/github.com/kata-containers/kata-containers/src/dragonball)
error: casting integer literal to `u64` is unnecessary
   --> src/kvm_context.rs:278:39
    |
278 |         let _ = c.create_vm_with_type(0 as u64).unwrap();
    |                                       ^^^^^^^^ help: try: `0_u64`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
    = note: `-D clippy::unnecessary-cast` implied by `-D warnings`

error: could not compile `dragonball` due to previous error
```

**Comment by fidencio** [comment]
Also, we're missing a `"Fixes: #"  in the commit message.

**Comment by studychao** [comment]
> lgtm, apart from a possible typo in one comment.
> 
> TDX CIs are failing due to network issues to download the crates needed to build the runtime-rs. It's important to also mention that while we're building the runtime-rs, we're not yet testing the runtime-rs with TDX, and that should be a follow-up work from your side (that I can guide you through, of course).

Got it! Thanks @fidencio , do you know how to restart the TDX CI?

File: src/dragonball/src/kvm_context.rs (hunk 1 above)
Comment (studychao) [reply src/dragonball/src/kvm_context.rs]: actually here should be `with` 😂
And I have fixed this.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @studychao!

The TDX QEMU Ci is failing due to spurious network.

**Pull request merged**